qiita_db/data.py

```diff
@@ -308,6 +329,81 @@ def create(cls, filetype, studies, filepaths=None):
 
         return rd
 
+    @classmethod
+    def delete(cls, raw_data_id, study_id):
+        """Removes the raw data with id raw_data_id
+
+        Parameters
+        ----------
+        raw_data_id : int
+            The raw data id
+        study_id : int
+            The study id
+
+        Raises
+        ------
+        QiitaDBUnknownIDError
+            If the raw data id doesn't exist
+        QiitaDBError
+            If the raw data is not linked to that study_id
+            If the raw data has prep templates associated
+        """
+        conn_handler = SQLConnectionHandler()
+
+        # check if the raw data exist
+        if not cls.exists(raw_data_id):
+            raise QiitaDBUnknownIDError(raw_data_id, "raw data")
+
+        study_raw_data_exists = conn_handler.execute_fetchone(
+            "SELECT EXISTS(SELECT * FROM qiita.study_raw_data WHERE "
+            "study_id = {0} AND raw_data_id = {1})".format(study_id,
+                                                           raw_data_id))[0]
+        if not study_raw_data_exists:
+            raise QiitaDBError(
+                "Raw data %d is not linked to study %d or the study "
+                "doesn't exist" % (raw_data_id, study_id))
+
+        # check if there are any prep templates for this study
+        prep_template_count = conn_handler.execute_fetchone(
```

Comment:
In addition, I don't think you are using the count, so probably with an `EXISTS` it will be enough.